cmd/termui/provider/logProvider.go

```diff
@@ -65,29 +69,26 @@ func openWebSocket(scheme string, address string, profile logger.Profile) (*webs
 	return conn, nil
 }
 
-// StartListeningOnWebSocket will listen if a new log message is received and will display it
-func StartListeningOnWebSocket(presenter PresenterHandler) {
-	go func() {
-		for {
-			msgType, message, err := webSocket.ReadMessage()
-			if msgType == websocket.CloseMessage {
-				return
-			}
-			if err == nil {
-				writeMessage(presenter, message)
-				continue
-			}
-
-			_, isConnectionClosed := err.(*websocket.CloseError)
-			if !isConnectionClosed {
-				log.Error("termui websocket error", "error", err.Error())
-			} else {
-				log.Debug("termui websocket terminated", "error", err.Error())
-			}
+// startListeningOnWebSocket will listen if a new log message is received and will display it
+func startListeningOnWebSocket(presenter PresenterHandler) {
+	for {
+		msgType, message, err := webSocket.ReadMessage()
+		if msgType == websocket.CloseMessage {
```

Comment:
if err != nil do the last part of the code - which has the return as well.

Reply:
It does. if err is not nil, then it will check the error's type, will log the error type and then there is a return